modules/core/infrastructure/persistence/session_repository.go

```diff
@@ -81,19 +91,53 @@ func (g *GormSessionRepository) Count(ctx context.Context) (int64, error) {
 	if err != nil {
 		return 0, err
 	}
+
+	tenant, err := composables.UseTenant(ctx)
+	if err != nil {
+		return 0, err
+	}
+
 	var count int64
-	if err := tx.QueryRow(ctx, sessionCountQuery).Scan(&count); err != nil {
+	if err := tx.QueryRow(ctx, sessionCountQuery+" WHERE tenant_id = $1", tenant.ID).Scan(&count); err != nil {
```

Comment:
For constructing sql queries in the runtime use methods from the `repo` package. Reference other usages of this package to get an idea on how to use it